app/helpers/injection_helper.rb

```diff
@@ -51,6 +51,18 @@ def inject_enterprise_attributes
     render partial: "json/injection_ams", locals: {name: 'enterpriseAttributes', json: "#{@enterprise_attributes.to_json}"}
   end
 
+  def inject_orders_by_distributor
+    # Convert ActiveRecord::Relation to array for serialization
+    data_array = spree_current_user.orders_by_distributor.to_a
+    data_array.each do |enterprise|
```

Comment:
Well I can't figure out how to do it in SQL/AR and I think it's very fiddly or not possible - I want to include show all 'completed' orders, but only show payments that were completed, not pending etc., as only completed payments count towards the order balance, which can be done by adding to the where clause. But even if there are no completed payments (no payments at all or only pending, void etc), we still need to show the order (with 'balance due' state). So we need to select orders where there are no payments that satisfy (state = complete), as well as orders and their completed payments for those that do have them. 

Hope that makes sense. My test data (a dump from aus staging) has a lot of orders with no completed payments - but maybe they wouldn't arise in real life? We could also show failed payments (or some types of failed payment) on the page, but would need to ensure that it's clear why they are not counting towards the balance.

Let me know if you have any ideas. In the meantime I moved the filtering code to the model in the latest commit.